packages/frontend/src/components/Explorer/CustomMetricModal/FilterForm.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+    ConditionalOperator,
+    createFilterRuleFromField,
+    FieldTarget,
+    FilterRule,
+    getFieldRef,
+    isDimension,
+} from '@lightdash/common';
+import { Button, Stack } from '@mantine/core';
+import { Dispatch, FC, SetStateAction, useCallback } from 'react';
+import { useExplorerContext } from '../../../providers/ExplorerProvider';
+import FilterRuleForm from '../../common/Filters/FilterRuleForm';
+import { useFiltersContext } from '../../common/Filters/FiltersProvider';
+import { addFieldRefToFilterRule } from './utils';
+
+export interface MetricFilterRuleWithFieldId
+    extends FilterRule<
+        ConditionalOperator,
+        FieldTarget & { fieldRef: string }
+    > {}
+
+export const FilterForm: FC<{
+    defaultFilterRuleFieldId: string;
+    customMetricFiltersWithIds: MetricFilterRuleWithFieldId[];
+    setCustomMetricFiltersWithIds: Dispatch<
+        SetStateAction<MetricFilterRuleWithFieldId[]>
+    >;
+}> = ({
+    defaultFilterRuleFieldId,
+    customMetricFiltersWithIds,
+    setCustomMetricFiltersWithIds,
+}) => {
+    const isEditMode = useExplorerContext(
+        (context) => context.state.isEditMode,
+    );
+    const { fieldsMap } = useFiltersContext();
+
+    const dimensions = Object.values(fieldsMap).filter(isDimension);
+
+    const addFieldRule = useCallback(() => {
+        const field = fieldsMap[defaultFilterRuleFieldId];
+
+        const newFilterRule = createFilterRuleFromField(field);
```

Comment:
We should handle the case where field is undefined. eg: the base dimension was removed or is missing.

Reply:
Yes yes, going to do that now!